src/apps/reservation-list/modal/delete-reservation/helper.ts

```diff
@@ -1,10 +1,36 @@
+import { UseMutateFunction } from "react-query";
+import { DeleteReservationsParams } from "../../../../core/fbs/model";
 import { UseTextFunction } from "../../../../core/utils/text";
 import { RequestStatus } from "../../../../core/utils/types/request";
 import {
   ReservationType,
   isDigitalReservation,
   isPhysicalReservation
 } from "../../../../core/utils/types/reservation-type";
+import { ApiResult } from "../../../../core/publizon/model";
+
+export type ParamsDigital = {
+  identifier: string;
+};
+export type ParamsPhysical = {
+  params?: DeleteReservationsParams;
+};
+export type OperationDigital = UseMutateFunction<
+  ApiResult | null,
+  unknown,
+  ParamsDigital,
+  unknown
+>;
+export type OperationPhysical = UseMutateFunction<
+  void | null,
+  unknown,
+  ParamsPhysical,
+  unknown
+>;
+type Request = {
+  params: { params: { reservationid: number[] } } | { identifier: string };
+  operation: OperationPhysical | OperationDigital;
+};
```

Comment:
```suggestion
type Request = {
  params: { params: { reservationid: number[] } } 
  operation: OperationPhysical
} | {
  params: { identifier: string }
  operation: OperationDigital
};
```

If we want to make types more precise consider matching parameter with operation type.

Reply:
Brilliant! Of course. Thx 👍